src/components/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -118,6 +118,28 @@ export const appendStatus = (
     } else if (mTask.status && !mTask.status.reason) {
       mTask.status.reason = RunStatus.Pending;
     }
+
+    // Determine any task test status
+    const taskRunName =
+      pipelineRun.status.taskRuns &&
+      Object.keys(pipelineRun.status.taskRuns).find(
+        (key) => pipelineRun.status.taskRuns[key].pipelineTaskName === mTask.name,
+      );
+    const taskRun = taskRunName && pipelineRun.status.taskRuns[taskRunName];
+    if (taskRun?.status?.taskResults) {
+      const testOutput = taskRun?.status?.taskResults.find(
+        (result) => result.name === 'HACBS_TEST_OUTPUT',
+      );
+      if (testOutput) {
+        try {
+          const outputValues = JSON.parse(testOutput.value);
+          mTask.testFailCount = outputValues.failures;
+          mTask.testWarnCount = outputValues.warnings;
```

Comment:
we need to also check for `result === FAILURE` before setting the testFailCount/testWarnCount because now the failed badge is showing even if the result is SUCCESS 

<img width="243" alt="Screenshot 2023-03-03 at 6 39 13 PM" src="https://user-images.githubusercontent.com/9964343/222728510-fdb45656-69cb-460b-96e6-cdff3263c466.png">